tests/unit/lms/services/canvas_test.py

```diff
@@ -13,271 +13,165 @@
-        # the file_id in our DB. Without a record we can't search for a
-        # matching file so we raise an error.
-
-        # There's no record of the file_id in the DB.
         file_service.get.return_value = None
 
-        with pytest.raises(CanvasFileNotFoundInCourse):
+        with pytest.raises(exception):
             public_url_for_file(
-                str(file_from_a_different_course["id"]), check_in_course=True
+                file_id=self.MISSING_FILE_ID, check_in_course=check_in_course
             )
 
-    def test_file_not_found_in_course_but_no_matching_file(
-        self, file_service, file_from_a_different_course, public_url_for_file
+    @pytest.mark.usefixtures("file_record")
+    @pytest.mark.parametrize(
+        "check_in_course,exception",
+        ((True, CanvasFileNotFoundInCourse), (False, CanvasAPIPermissionError)),
+    )
+    def test_remapping_raises_if_theres_no_match_in_the_course(
+        self, canvas_service, public_url_for_file, check_in_course, exception
     ):
-        # This is what happens when a teacher launches an assignment whose
-        # file_id is *not* in the current course and even though we do have a
-        # record of this file_id in our DB we don't find a matching file in the
-        # current course. Since we can't find a matching file we can't fix the
-        # assignment so we raise an error.
-
-        # The file record that we find in our DB. Its name doesn't match any
-        # file in the current course.
-        file_service.get.return_value = factories.File(name="foo")
+        canvas_service.api.list_files.return_value = []
 
-        with pytest.raises(CanvasFileNotFoundInCourse):
+        with pytest.raises(exception):
             public_url_for_file(
-                file_id=str(file_from_a_different_course["id"]), check_in_course=True
+                file_id=self.MISSING_FILE_ID, check_in_course=check_in_course
             )
 
-    def test_permissions_error_but_no_file_info(
-        self,
-        canvas_service,
-        file_service,
-        file_from_a_different_course,
-        public_url_for_file,
-    ):
-        # This is what happens when a student launches an assignment whose
-        # file_id is *not* in the current course and we don't have a record of
-        # the file_id in our DB. Without a record we can't search for a
-        # matching file so we raise an error.
-        canvas_service.api.public_url.side_effect = CanvasAPIPermissionError
-        file_service.get.return_value = None
-
-        with pytest.raises(CanvasAPIPermissionError):
-            public_url_for_file(file_id=str(file_from_a_different_course["id"]))
+    @pytest.fixture(autouse=True)
+    def canvas_api_client(self, canvas_api_client):
+        def public_url(file_id):
+            # Rig public_url to blow if we get MISSING_FILE_ID
+            if file_id == self.MISSING_FILE_ID:
+                raise CanvasAPIPermissionError()
```

Comment:
This really helped simplify and unify a lot of scenarios. Now the API just fails if it ever gets called with the bad id.

This means that the `check_in_course` true and false branches do not require different setup (as far as this API client goes).